lib/Service/StackService.php

```diff
@@ -362,4 +379,110 @@ public function reorder($id, $order) {
 
 		return $result;
 	}
+
+	/**
+	 * @param $id
+	 * @param $boardId
+	 * @param $userId
+	 * @return Stack
+	 * @throws StatusException
+	 * @throws BadRequestException
+	 */
+	public function clone($id, $boardId, $userId) {
+		if (is_numeric($id) === false) {
+			throw new BadRequestException('stack id must be a number');
+		}
+		if (is_numeric($boardId) === false) {
+			throw new BadRequestException('board id must be a number');
+		}
+
+		$this->permissionService->checkPermission(null, $boardId, Acl::PERMISSION_MANAGE);
```

Comment:
Any specific reason for requiring manage permissions and not just write permissions?

Reply:
@juliushaertl 
You mean "PERMISSION_EDIT", right?

Reply:
Well this is not fixed yet. We will should switch this to edit permission.

Reply:
Ah wait, we indeed require manage permissions for all stack operations, so this is ok then.